### PR TITLE
Add check for property

### PR DESCRIPTION
**Pull request by jdposthuma**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

This PR fixes the open issue.

Related issue: #7416

### Approach

Simply added a truthy check 

### TODOs before merging

- [x] Add test cases
- [x] Add entry to changelog

### CODE REVIEW

**Comment by dplewis** [comment]
@jdposthuma Thanks for the contribution. Can you go though the ToDo list? Add a test case, update the change log, etc?

**Comment by jdposthuma** [comment]
> @jdposthuma Thanks for the contribution. Can you go though the ToDo list? Add a test case, update the change log, etc?

@dplewis - Done. I've added to the changelog.

None of the other items seemed valid:
 - Add test cases

I haven't been able to capture (or simulate) the failing class

- Add changes to documentation (guides, repository pages, in-code descriptions)

Too insignificant to add to docs
 
- Add security check

Doesn't change security
 
- Add new Parse Error codes to Parse JS SDK

No new parse error code

**Comment by mtrezza** [comment]
Except for the test case, you can delete the non-applicable TODOs from the list.

The change makes sense, even if there wasn't a reported issue, just for code quality. However, it would be good to have a test case, to make your fix sustainable and prevent a future change from reverting your fix by mistake. And to verify that it solved the issue.

**Comment by mtrezza** [comment]
@jdposthuma Can you please rebase on master?

**Comment by jdposthuma** [comment]
> @jdposthuma Can you please rebase on master?

Done

**Comment by jdposthuma** [comment]
@mtrezza - Test case added. 

I'm really glad you pushed for it because my original fix had a logic problem. 🙏

**Comment by jdposthuma** [comment]
Hi @mtrezza - do I need to do anything else to get this to merge?

File: CHANGELOG.md [hunk 1]
```diff
@@ -134,6 +134,7 @@ ___
 - Add NPM package-lock version check to CI (Manuel Trezza) [#7333](https://github.com/parse-community/parse-server/pull/7333)
 - Fix incorrect LiveQuery events triggered for multiple subscriptions on the same class with different events [#7341](https://github.com/parse-community/parse-server/pull/7341)
 - Fix select and excludeKey queries to properly accept JSON string arrays. Also allow nested fields in exclude (Corey Baker) [#7242](https://github.com/parse-community/parse-server/pull/7242)
+- Added safety check for attribute when filtering (Jason Posthuma) [#7421](https://github.com/parse-community/parse-server/pull/7421)
```
Comment (mtrezza) [inline CHANGELOG.md:137]: Can you please make this more specific, for example:

> - Fix LiveQuery server crash when using `$all` query operator on a missing object key (Jason Posthuma) [#7421](https://github.com/parse-community/parse-server/pull/7421)

**Review by mtrezza — CHANGES_REQUESTED** [review]
Just the changelog entry

File: CHANGELOG.md (hunk 1 above)
Comment (jdposthuma) [reply CHANGELOG.md]: Done

**Review by mtrezza — APPROVED** [review]
Looks good to me, thanks for this PR!

**Review by davimacedo — APPROVED** [review]
LGTM!

**Comment by jdposthuma** [comment]
Thanks guys! When's the next release planned?

**Comment by mtrezza** [comment]
The next release has no fixed date, but it is estimated to occur later this year. The reason is that we are working on defining a [release cycle](https://github.com/parse-community/parse-server/issues/7271) with release automation that should bring a significant improvement in terms of release frequency and predictability. In addition, we are aiming to introduce an alpha / beta distribution channel let releases undergo testing and become more robust before official release. Currently, the master branch contains an unusual high number of breaking and conceptual changes some of which have not been thoroughly tested. Therefore - in any case - we aim to let this go though a beta distribution first to gather feedback.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**